CMakeLists.txt

```diff
@@ -52,6 +52,9 @@ IF(WIN32)
 ELSE(WIN32)
   ADD_COMPILE_OPTIONS (-Wextra -Wno-unused-parameter)
 ENDIF(WIN32)
+IF(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+  set (CMAKE_SHARED_LINKER_FLAGS "-undefined dynamic_lookup")
+ENDIF()
```

Comment:
Thanks for the sug.  Would it be more idiomatic to write this as:

```
IF(APPLE)
   ...
ENDIF()
```
?

Looks good otherwise.  If that construct works for you too then please update it and we'll get this merged in.

Reply:
thanks for the hint. Made the change as requested.